src/control/xojfile/LoadHandler.cpp

```diff
@@ -378,7 +381,8 @@ void LoadHandler::parseBgPixmap() {
 
         this->page->setBackgroundImage(img);
     } else if (!strcmp(domain, "clone")) {
-        PageRef p = doc.getPage(stoull(filepath.string()));
+        size_t nr = static_cast<size_t>(stoull(filename));
```

Comment:
If `filename` is an invalid number (which is possible if the file is ill-formed), then `stoull` will return 0. I think [`g_ascii_strtoull`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-ascii-strtoull) would be a better option because it lets you check for invalid values.

Reply:
Turns out you are right about the conversion to path and converting back to string. I changed it back.

I have now introduced checking for invalid values using `g_ascii_strtoull`